driver-core/src/main/com/mongodb/internal/connection/DefaultServerMonitor.java

```diff
@@ -178,6 +179,10 @@ public void run() {
                     }
                     waitForNext();
                 }
+            } catch (MongoInterruptedException e) {
+                // ignore
+            } catch (RuntimeException e) {
+                LOGGER.error(format("Server monitor for %s exiting with exception", serverId), e);
```

Comment:
In addition to catching and ignoring `MongoInterruptedException`, it seems better to use our own logger to log any other exceptions rather than rely on the default behavior of threads when their Runnable exits exceptionally.